flink-connectors/flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/RMQSource.java

```diff
@@ -151,7 +151,9 @@ public void open(Configuration config) throws Exception {
 			if (channel == null) {
 				throw new RuntimeException("None of RabbitMQ channels are available");
 			}
-			setupQueue();
+			if (rmqConnectionConfig.hasToCreateQueueOnSetup()) {
```

Comment:
Acceptable in devel environment, in production the application should not be able to operate not-owned objects and creating an extended class to avoid the queueDeclare on start seems to be an heavy solution. However, we have a solution and it's complies to our needed. So we modify to your proposal.